### PR TITLE
feat(migration): remove parallel zkt iteration code

### PR DESCRIPTION
**Pull request by 0xbenyun**
# Description

I removed parallel zkt iteration code to improve stability

### CODE REVIEW

File: migration/migrator.go [hunk 1]
```diff
@@ -215,28 +215,29 @@ func (m *StateMigrator) migrateStorage(
...
 
 		slots.Add(1)
-		log.Trace("Updated storage slot to MPT", "contract", address.Hex(), "index", common.BytesToHash(key).Hex())
-		return nil
-	})
-	if err != nil {
-		return common.Hash{}, err
+		log.Trace("Updated storage slot to MPT", "contract", address.Hex(), "index", common.BytesToHash(iter.Key).Hex())
+	}
+	if iter.Err != nil {
```
Comment (sm-stack) [inline migration/migrator.go:239]: Question here, if the above for loop ends, won't this `iter.Err` be always not nil? Is some kind of adding a handling for `errIteratorEnd` needed here?

**Comment by seolaoh** [comment]
Could you resolve the conflict?

File: migration/migrator.go (hunk 1 above)
Comment (0xbenyun) [reply migration/migrator.go:239]: if the loop ends as expected, iter.Err will be nil. However, if the loop ends unexpectedly, iter.Err will contain a seekError

**Review by sm-stack — APPROVED** [review]

File: migration/migrator.go [hunk 2]
```diff
@@ -215,28 +215,29 @@ func (m *StateMigrator) migrateStorage(
...
 	}
 
-	var mu sync.Mutex
+	nodeIt, err := zkt.NodeIterator(nil)
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("failed to open node iterator (root: %s): %w", zkt.Hash(), err)
+	}
+	iter := trie.NewIterator(nodeIt)
 	var slots atomic.Uint64
-	err = hashRangeIterator(m.ctx, zkt, NumProcessStorage, func(key, value []byte) error {
```
Comment (seolaoh) [inline migration/migrator.go:220]: I think by removing `hashRangeIterator` which contains checking the `m.ctx.Done()` signal, if `m.ctx.Done()` signal arrives while executing `migrateStorage`, `migrateAccount` will wait until `migrateStorage` execution completed. How about adding L131-L142 goroutine in `migrateStorage` function also?

File: migration/migrator_validate.go [hunk 3]
```diff
@@ -22,120 +20,118 @@ func (m *StateMigrator) ValidateStateWithIterator(mptRoot common.Hash, zkRoot co
...
+			return fmt.Errorf("account %s root should be empty root hash. got %s", addr, mptAcc.Root)
+		}
 
-			accounts.Add(1)
-			return nil
-		})
+		accounts.Add(1)
 		if err != nil {
 			return err
 		}
```
Comment (seolaoh) [inline migration/migrator_validate.go:131]: This is meaningless

**Review by kangsorang — APPROVED** [review]

File: migration/migrator.go (hunk 2 above)
Comment (0xbenyun) [reply migration/migrator.go:220]: Oh, do you mean adding logs for the migrated storage number in the goroutine, or halting the iteration process immediately when m.ctx.Done() is received, by using a goroutine?

Comment (seolaoh) [reply migration/migrator.go:220]: I meant that if `m.ctx.Done()` is received, `migrateAccount` and `migrateStorage` should also return immediately. You are right, the current goroutine returns the logging only. Then how about adding below code block to below L179?
```go
		select {
		case <-m.ctx.Done():
			return m.ctx.Err()
		default:
		}
```

Comment (0xbenyun) [reply migration/migrator.go:220]: Good idea. I’ve applied it.

**Review by seolaoh — APPROVED** [review]

**Review by Pangssu — APPROVED** [review]

**Pull request merged**